File: Liive Ride Design System/handoff/android/LiiveBottomSheet.kt
//  LiiveBottomSheet.kt  ·  Liive Ride DS (Compose)  ·  mirrors components/ride/BottomSheet
//  The sheet content container that rises over the map. For a real modal use
//  Material3 ModalBottomSheet and place this content inside; this composable
//  is the styled body (rounded top, grabber, opaque surface, nav-bar inset).
package com.liive.ride.designsystem

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

@Composable
fun LiiveBottomSheet(
    modifier: Modifier = Modifier,
    grabber: Boolean = true,
    padding: Dp = 16.dp,
    content: @Composable ColumnScope.() -> Unit
) {
    val c = LiiveTheme.colors
    Column(
        modifier
            .fillMaxWidth()
            .clip(LiiveRadius.sheetTop)
            .background(c.surfaceSheet)
            .padding(horizontal = padding)
            .padding(top = if (grabber) 8.dp else padding)
            .padding(bottom = padding)
            .navigationBarsPadding(),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        if (grabber) {
            Box(Modifier.padding(bottom = 14.dp).size(width = 36.dp, height = 5.dp)
                .clip(CircleShape).background(c.fill))
        }
        Column(Modifier.fillMaxWidth(), content = content)
    }
}

File: Liive Ride Design System/handoff/ios/LiiveBottomSheet.swift
//  LiiveBottomSheet.swift  ·  Liive Ride DS (SwiftUI)
//  The sheet container that rises over the map: rounded top, grabber,
//  opaque surface, safe-area bottom inset. Drop your screen content inside.

import SwiftUI

public struct LiiveBottomSheet<Content: View>: View {
    var grabber: Bool = true
    var padding: CGFloat = 16
    let content: Content

    public init(grabber: Bool = true, padding: CGFloat = 16, @ViewBuilder content: () -> Content) {
        self.grabber = grabber; self.padding = padding; self.content = content()
    }

    public var body: some View {
        VStack(spacing: 0) {
            if grabber {
                Capsule().fill(LiiveColor.fill)
                    .frame(width: 36, height: 5)
                    .padding(.top, 8).padding(.bottom, 14)
            }
            content
        }
        .frame(maxWidth: .infinity)
        .padding(.horizontal, padding)
        .padding(.bottom, padding + LiiveSpacing.safeBottom)
        .padding(.top, grabber ? 0 : padding)
        .background(LiiveColor.surfaceSheet)
        .clipShape(TopRoundedRectangle(radius: LiiveRadius.xxxl))
        .liiveShadow(.sheet)
    }
}

/// Rounds only the top two corners.
struct TopRoundedRectangle: Shape {
    var radius: CGFloat
    func path(in rect: CGRect) -> Path {
        Path(UIBezierPath(roundedRect: rect, byRoundingCorners: [.topLeft, .topRight],
                          cornerRadii: CGSize(width: radius, height: radius)).cgPath)
    }
}
